src/main/java/io/github/moulberry/notenoughupdates/util/TwoKeyCache.java

```diff
@@ -0,0 +1,15 @@
+package io.github.moulberry.notenoughupdates.util;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Supplier;
+
+public class TwoKeyCache<K1, K2, V> {
+
+	private final Map<K1, Map<K2, V>> cache = new HashMap<>();
```

Comment:
I feel like a `Map<Pair<K1,K2>, V>` makes more sense here (esp. considering the minimum size of a hash map).

Reply:
fair